gateway-sp-comms/src/ereport.rs

```diff
@@ -0,0 +1,831 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2025 Oxide Computer Company
+
+use crate::error::CommunicationError;
+use crate::error::EreportError;
+use crate::shared_socket;
+use crate::single_sp;
+use crate::SpRetryConfig;
+use async_trait::async_trait;
+pub use gateway_messages::ereport::Ena;
+use gateway_messages::ereport::EreportRequest;
+use gateway_messages::ereport::EreportResponseHeader;
+use gateway_messages::ereport::RequestV0;
+use gateway_messages::ereport::ResponseHeaderV0;
+pub use gateway_messages::ereport::RestartId;
+use serde::Deserialize;
+use serde_cbor::Value as CborValue;
+use serde_json::Value as JsonValue;
+use slog::debug;
+use slog::error;
+use slog::trace;
+use slog::warn;
+use slog_error_chain::SlogInlineError;
+use std::collections::BTreeMap;
+use std::num::NonZeroU8;
+use thiserror::Error;
+use tokio::sync::mpsc;
+use tokio::sync::oneshot;
+
+pub const SP_PORT: u16 = 0xDEAD;
+pub const MGS_PORT: u16 = 0xDEAF;
```

Comment:
Tiny nit - could we write these in decimal? Most OS networking tools deal with ports in decimal, and I've occasionally grepped through codebases to pick out port numbers, and these being in hex would be a bit of a stumbling block.